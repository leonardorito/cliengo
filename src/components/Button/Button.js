import React from "react";
import "./Button.scss";

const Button = ({ text, onClick }) => {
  return (
    <button className="Button" onClick={onClick}>
      {text}
    </button>
  );
};

export default Button;
