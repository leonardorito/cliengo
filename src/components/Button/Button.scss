@import '../../variables';

.Button {
  background-color: $green;
  color: $white;
  border: 0;
  box-shadow: 0px 3px 6px $shadow-toggle;
  border-radius: 16px;
  width: 86px;
  height: 32px;
}
