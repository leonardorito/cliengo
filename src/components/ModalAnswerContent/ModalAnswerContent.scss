@import "../../variables";

.ModalAnswerContent {
  h4 {
		margin: 0;
		overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  p {
    margin-top: 0;
    margin-bottom: 15px;
  }

  .modal-img {
    width: 13px;
    height: 13px;
    margin-right: 5px;
  }

  .close {
    width: 14px;
		height: 14px;
		margin-left: 20px;
		cursor: pointer;
  }

  .modal-title {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 15px;
  }

  textarea {
    width: 100%;
    resize: none;
    border: 1px solid $disabled-gray;
    border-radius: 4px;
    padding: 7px 12px;
    font-size: 14px;
    font-family: "Nunito", sans-serif;
  }

  .options {
    margin-top: 20px;
    text-align: center;
  }
}
