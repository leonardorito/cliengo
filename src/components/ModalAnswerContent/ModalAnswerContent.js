import React, { Fragment, useState } from "react";
import "./ModalAnswerContent.scss";
import Button from "../Button/Button";
import SwitchToggle from "../SwitchToggle/SwitchToggle";

const ModalAnswerContent = ({ save, data, close }) => {
  const [description, setDescription] = useState(data.description);
  const [active, setActive] = useState(data.active);

  const sendNewData = () => {
    save({
      ...data,
      description,
      active: !active,
    });
  };

  return (
    <Fragment>
      <div className="ModalAnswerContent">
        <div className="modal-title">
          <h4>
            <img src="/assets/flash.svg" alt="Rapido" className="modal-img" />
            <span className="purple-text">Respuesta a: </span>{data.title}
          </h4>
          <div className="d-flex">
            <SwitchToggle
              id={`modal-${data.id}`}
              isOn={active}
              onColor="#1ABC9C"
              handleToggle={() => setActive(!active)}
            />
						<img src="/assets/close.svg" alt="Cerrar" className="close" onClick={close} />
          </div>
        </div>
        <p>
          <b>
            Si el cliente consulta por {data.title.toLowerCase()}, el chatbot
            responder√°
          </b>
        </p>
        <textarea
          rows="2"
          value={description}
          onChange={(el) => setDescription(el.target.value)}
        ></textarea>
        <div className="options">
          <Button text="Guardar" onClick={sendNewData} />
        </div>
      </div>
    </Fragment>
  );
};

export default ModalAnswerContent;
