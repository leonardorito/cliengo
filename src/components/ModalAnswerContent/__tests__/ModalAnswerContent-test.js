import React from "react";
import cardData from "../../../constants/cards-data";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ModalAnswerContent from "../ModalAnswerContent";
import Button from "../../Button/Button";

it("ModalAnswerContent should render correctly", () => {
  const data = cardData;
  const save = jest.fn();
  const close = jest.fn();
  const wrapper = mount(
    <ModalAnswerContent data={data[0]} save={save} close={close} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("Textarea should have description text", () => {
  const data = cardData;
  const save = jest.fn();
  const close = jest.fn();
  const wrapper = mount(
    <ModalAnswerContent data={data[0]} save={save} close={close} />
  );
  expect(wrapper.find("textarea").props().value).toBe(data[0].description);
});

it("Button onClick works", () => {
  const mockCallBack = jest.fn();

  const button = shallow(<Button text="Guardar" onClick={mockCallBack} />);
  button.find("button").simulate("click");
  expect(mockCallBack.mock.calls.length).toEqual(1);
});
