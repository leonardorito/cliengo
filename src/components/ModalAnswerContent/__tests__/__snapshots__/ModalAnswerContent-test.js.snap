// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAnswerContent should render correctly 1`] = `
<ModalAnswerContent
  close={[MockFunction]}
  data={
    Object {
      "active": true,
      "description": "Atendemos de lunes a viernes de 9 a 18 hs y sábados de 10 a 13hs.",
      "id": 1,
      "title": "Horario de atención",
    }
  }
  save={[MockFunction]}
>
  <div
    className="ModalAnswerContent"
  >
    <div
      className="modal-title"
    >
      <h4>
        <img
          alt="Rapido"
          className="modal-img"
          src="/assets/flash.svg"
        />
        <span
          className="purple-text"
        >
          Respuesta a: 
        </span>
        Horario de atención
      </h4>
      <div
        className="d-flex"
      >
        <SwitchToggle
          handleToggle={[Function]}
          id="modal-1"
          isOn={true}
          onColor="#1ABC9C"
        >
          <div
            className="SwitchToggle"
          >
            <input
              checked={true}
              className="react-switch-checkbox"
              id="react-switch-modal-1"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="checkbox"
            />
            <label
              className="react-switch-label"
              htmlFor="react-switch-modal-1"
              style={
                Object {
                  "background": "#1ABC9C",
                }
              }
            >
              <span
                className="react-switch-button"
              />
            </label>
          </div>
        </SwitchToggle>
        <img
          alt="Cerrar"
          className="close"
          onClick={[MockFunction]}
          src="/assets/close.svg"
        />
      </div>
    </div>
    <p>
      <b>
        Si el cliente consulta por 
        horario de atención
        , el chatbot responderá
      </b>
    </p>
    <ResizableTextArea
      changeValue={[Function]}
      value="Atendemos de lunes a viernes de 9 a 18 hs y sábados de 10 a 13hs."
    >
      <div
        className="TextArea"
      >
        <textarea
          onChange={[Function]}
          rows={2}
          value="Atendemos de lunes a viernes de 9 a 18 hs y sábados de 10 a 13hs."
        />
      </div>
    </ResizableTextArea>
    <div
      className="options"
    >
      <Button
        onClick={[Function]}
        text="Guardar"
      >
        <button
          className="Button"
          onClick={[Function]}
        >
          Guardar
        </button>
      </Button>
    </div>
  </div>
</ModalAnswerContent>
`;
